src/validation/__tests__/OverlappingFieldsCanBeMergedRule-test.ts

```diff
@@ -1138,4 +1138,70 @@ describe('Validate: Overlapping fields can be merged', () => {
       },
     ]);
   });
+
+  it('does not hit stack size limits', () => {
+    const n = 10000;
```

Comment:
The n is purposefully high here because some browsers cap at 9.000 and some environments at 1.000